Marlin/stepper.cpp

```diff
@@ -796,6 +819,104 @@ void Stepper::isr() {
   #endif
 }
 
+void Stepper::chunk_steps() {
+  const uint8_t *cur_chunk = chunk_buffer[current_block->chunk_idx];
+  const uint8_t cur_chunk_block = uint8_t(step_events_completed >> 3) << 1;
+  const uint8_t block_steps = uint8_t(step_events_completed & 0x7);
+
+  const uint8_t a = cur_chunk[cur_chunk_block];
+  const uint8_t b = cur_chunk[cur_chunk_block + 1];
+
+  const uint8_t dX = a >> 4;
+  const uint8_t dY = a & 0xF;
+  const uint8_t dZ = b >> 4;
+  const uint8_t dE = b & 0xF;
+
+  uint8_t steps[4] = { 0 };
+  steps[X_AXIS] = block_moves[dX][(block_steps + 0) & 0x7];
```

Comment:
The 0,2,4,6 offsets here are to just add a permanent offset to the wave tables for each axis. Probably not needed and just wasting cycles, but the theory there was to prevent noise (if all axis are stepping at the same rate, we want the pulses to stagger a bit)